src/components/gltf-model.js

```diff
@@ -26,6 +26,8 @@ module.exports.Component = registerComponent('gltf-model', {
       self.model.animations = gltfModel.animations;
       el.setObject3D('mesh', self.model);
       el.emit('model-loaded', {format: 'gltf', model: self.model});
+    }, undefined /* onProgress */, function gltfFailed () {
+      el.emit('model-error', {format: 'gltf', src: src});
```

Comment:
We probably should throw an error instead of an event? An similar scenario can be found in the `text` component: https://github.com/aframevr/aframe/blob/master/src/components/text.js#L391

Reply:
Could do both? I listen to the error message to try loading a different (fallback) asset file, I think it could be useful in other contexts for the caller to know that the file load failed.

Reply:
yes, we could have both

Reply:
+1 for both. The errors should be helpful for developers debugging, who may not know to listen for the event.

But my inclination would be to log an error or warning, through the debug util, rather than throwing something that can't be caught. See [`obj-model`](https://github.com/aframevr/aframe/blob/master/src/components/obj-model.js#L5).

Reply:
Cool! Let's add both then 👌

Reply:
Will do both, thanks!